### PR TITLE
Updating the user guide for the qmmm module

### PR DESCRIPTION
**Pull request by hczhai**
QM/MM user guide.

### CODE REVIEW

File: source/user/qmmm.rst [hunk 1]
```diff
@@ -1,5 +1,96 @@
...
 *************
 
 *Modules*: :mod:`qmmm`
+
+Introduction
+============
+
+In the Quantum Mechanics - Molecular Mechanics (QM/MM) method,
+the system is divided into a QM region and an MM region.
+Different QM/MM models mainly differ in the treatment of the interaction
```
Comment (susilehtola) [inline source/user/qmmm.rst]:
```suggestion
QM/MM models mainly differ in the treatment of the interaction
```

File: source/user/qmmm.rst [hunk 2]
```diff
@@ -1,5 +1,96 @@
...
+
+Another QM/MM model - polarizable embedding model - is provided in the
+:mod:`solvent` module. See :ref:`user_solvent` for details.
+
+Usage
+=====
+
+Functions implemented in the :mod:`qmmm` module for MM background charges support
+all molecular electronic structure methods
+by decorating the underlying SCF methods with :func:`qmmm.mm_charge`.
```
Comment (susilehtola) [inline source/user/qmmm.rst]: Unclear, please reword.

File: source/user/qmmm.rst [hunk 3]
```diff
@@ -1,5 +1,96 @@
...
+    >>> charges = [-0.3]
+    >>> mf = qmmm.mm_charge(scf.RHF(mol), coords, charges)
+    >>> mf.kernel()
+    converged SCF energy = -100.045455504517
+
+In the above example, the coordinates (in the same unit as ``mol.unit``) and
+charges of the point charges in the MM region are given by ``coords`` and ``charges``,
+respectively.
+
+Analytical nuclear gradients will be calculated with the background charges. ::
```
Comment (susilehtola) [inline source/user/qmmm.rst]:
```suggestion
Analytical nuclear gradients are calculated with the background charges. ::
```

File: source/user/qmmm.rst [hunk 4]
```diff
@@ -1,5 +1,96 @@
...
+    ----------------------------------------------
+
+.. note::
+
+    The gradients obtained from ::
+
+    >>> from pyscf import grad
+    >>> grad.RHF(mf).run()
+
+    for a :func:`qmmm.mm_charge` decorated ``mf`` object will be incorrect.
```
Comment (susilehtola) [inline source/user/qmmm.rst]: .. because they will be missing the contributions from XX

File: source/user/qmmm.rst (hunk 2 above)
Comment (hczhai) [reply source/user/qmmm.rst]: Should be fixed in this commit: 4f090e17f2669f9c863f466718f601e722c98142
Supported SCF methods are explicitly listed, and a note is given for the unsupported `scf.GHF` and `scf.GKS`.

File: source/user/qmmm.rst (hunk 4 above)
Comment (hczhai) [reply source/user/qmmm.rst]: Updated in this commit: 4f090e17f2669f9c863f466718f601e722c98142

**Review by susilehtola — APPROVED** [review]

**Pull request merged**